packages/sales-api-service/src/services/contacts.service.js

```diff
@@ -28,24 +28,37 @@ const debug = db('sales:transformers')
  * @param {!ContactPayload} payload The payload to be transformed
  * @returns {Promise<Contact>}
  */
-export const resolveContactPayload = async payload => {
+export const resolveContactPayload = async (permission, payload) => {
   const { id, country, preferredMethodOfConfirmation, preferredMethodOfNewsletter, preferredMethodOfReminder, ...primitives } = payload
 
   const contactInCRM = await findContactInCRM(payload)
   const contact = Object.assign(contactInCRM || new Contact(), primitives)
 
-  contact.preferredMethodOfConfirmation = await getGlobalOptionSetValue(
-    Contact.definition.mappings.preferredMethodOfConfirmation.ref,
-    preferredMethodOfConfirmation
+  contact.preferredMethodOfReminder = await getGlobalOptionSetValue(
+    Contact.definition.mappings.preferredMethodOfReminder.ref,
+    preferredMethodOfReminder
   )
   contact.preferredMethodOfNewsletter = await getGlobalOptionSetValue(
     Contact.definition.mappings.preferredMethodOfNewsletter.ref,
     preferredMethodOfNewsletter
   )
-  contact.preferredMethodOfReminder = await getGlobalOptionSetValue(
-    Contact.definition.mappings.preferredMethodOfReminder.ref,
-    preferredMethodOfReminder
+  contact.shortTermPreferredMethodOfConfirmation = await getGlobalOptionSetValue(
+    Contact.definition.mappings.shortTermPreferredMethodOfConfirmation.ref,
+    preferredMethodOfConfirmation
   )
+
+  if (permission.licenceLength !== '12M' && contactInCRM) {
+    contact.preferredMethodOfConfirmation = await getGlobalOptionSetValue(
+      Contact.definition.mappings.preferredMethodOfConfirmation.ref,
+      contactInCRM.preferredMethodOfConfirmation.description
+    )
+  } else {
+    contact.preferredMethodOfConfirmation = await getGlobalOptionSetValue(
+      Contact.definition.mappings.shortTermPreferredMethodOfConfirmation.ref,
+      preferredMethodOfConfirmation
+    )
+  }
```

Comment:
I'm not sure your logic for assigning the short term method of confirmation is right. You are _always_ setting it to preferredMethodOfConfirmation, whether or not it's a short term or long term licence.
Then if it's a short term licence and the contact is in the CRM, you set the long term method of confirmation to be whatever it already is. You're essentially doing this:
```
contact.preferredMethodOfConfirmation = contact.preferredMethodOfConfirmation
```
I think you need to have a decision based on whether it's an existing contact or not, and _then_ set the values, perhaps something like this:
```
const contactAlreadyExists = (contactInCRM === undefined) // n.b. check what findContactInCRM returns if there isn't one, it might not be undefined
if (contactAlreadyExists) {
  if (shortTermLicence) {
    contact.shortTermPreferredMethodOfConfirmation = preferredMethodOfConfirmation
  } else {
    contact.preferredMethodOfConfirmation = preferredMethodOfConfirmation
  }
} else {
    contact.shortTermPreferredMethodOfConfirmation = preferredMethodOfConfirmation
    contact.preferredMethodOfConfirmation = preferredMethodOfConfirmation
}
```
It might actually be that this decision logic is unnecessary, and you can actually do this:
```
  if (shortTermLicence) {
    contact.shortTermPreferredMethodOfConfirmation = preferredMethodOfConfirmation
  } else {
    contact.preferredMethodOfConfirmation = preferredMethodOfConfirmation
  }
```
You'll have to give it a try. If the latter is the case, then you don't need the `contactAlreadyExists` variable in the first block.

Reply:
think was agreed if short term licence it would be set regardless, so if long term licence is bought it would overwrite the short term confirmation as default

Reply:
I'm still not convinced by the logic you've put in.
The AC says: "If the permission is for a 12 month licence, and the method of confirmation differs from the Preferred Method Of Confirmation on the Contact ID, then the latter will be update to match the Permissions confirmation method."

We don't need to check whether the method of confirmation differs, that'd be unnecessary complication, but we need to set the (long term) preferred method of confirmation if it's a long term licence or the contact doesn't exist in the CRM

So instead of the if...else, you'd do:
```
if (permission.licenceLength === '12M' || contactInCRM === undefined) {
    contact.preferredMethodOfConfirmation = await getGlobalOptionSetValue(
       Contact.definition.mappings.shortTermPreferredMethodOfConfirmation.ref,
       preferredMethodOfConfirmation
     )
}
```
Let's have a chat about this in any case